### PR TITLE
Ports-Elise

### PR DESCRIPTION
**Pull request by ChubbyCub**
# Task List
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
Describe in your own words what the Model is doing in Rails | Model contains data for the application
Describe in your own words what the Controller is doing in Rails | Controller handles HTTPS requests and gather data from the models
Describe in your own words what the View is doing in Rails | View renders HTML
Describe an edge-case controller test you wrote |I tested the update action when given an invalid id. My website is expected to redirect the users to the homepage in this case.
What is the purpose of using strong params? (i.e. the params method in the controller) | Strong params help DRY up our codes. 
How are Rails migrations related to Rails models? | Rails migrations can start a new database or modify the existing database for the models.
Describe one area of Rails that are still unclear on |I am still unclear on how to know the exact structure of task_params. Why do we expect the structure {task: {}} ? Dee mentioned it might be different for other situations. Which situation is it?

### CODE REVIEW

**Comment by tildeee** [comment]
# Task List
## What We're Looking For

Feature | Feedback
:------------- | :-------------
**Baseline** |
Appropriate Git Usage with no extraneous files checked in | x
Answered comprehension questions |
Successfully handles: Index, Show | x
Index & Show tests pass | x
Successfully handles: New, Create | x
New & Create tests pass | x, I get one failing test for the create method. I think this is due to the date object in `due_date` not being in the same format
Successfully handles: Edit, Update | x
Tests for Edit & Update test valid & invalid task ids | x
Successfully handles: Destroy, Task Complete | s
Tests for Destroy & Task Complete include tests for valid and invalid task ids | x
Routes follow RESTful conventions | x
Uses named routes (like `_path`) | x
**Overall** |



Well done on this project, Elise! Your project looks great and meets all of the requirements: Rails best practices, RESTful routing, and CRUD operations. Your controller testing is well-written. And your app looks great, too!


When I run your tests, I get a failing test for one of the create method: the `due_date` is not matching. I think this is simply a formatting issue.


Also, to answer your question in the comprehension questions: The structure of `params` where `task` is nested comes from how Rails's `form_with` automatically sends back the structure. A scenario where this structure would look different is if you didn't use `form_with`, but a different form


That being said, super great work! Well done!

File: app/controllers/tasks_controller.rb [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+      task.update(
+        completed_date: DateTime.now,
+        completed: true,
+      )
+    else
+      task.update(
+        completed_date: nil,
+        completed: false,
+      )
+    end
```
Comment (tildeee) [inline app/controllers/tasks_controller.rb:63]: Well done! This is one of my ideal ways of handling this logic. Very elegant!

File: app/views/tasks/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,27 @@
+<link rel="stylesheet" href="../assets/stylesheets/normalize.css">
```
Comment (tildeee) [inline app/views/tasks/index.html.erb:1]: You shouldn't need to add this file here -- we haven't gone over how to have different stylesheets, but in the future, it will be included
